src/Features/CSharp/Portable/Completion/KeywordRecommenders/AssemblyKeywordRecommender.cs

```diff
@@ -17,15 +15,21 @@ public AssemblyKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            if (context.IsTypeAttributeContext(cancellationToken))
+            var token = context.TargetToken;
+
+            if (token.Kind() != SyntaxKind.OpenBracketToken)
             {
-                var token = context.LeftToken;
-                var type = token.GetAncestor<MemberDeclarationSyntax>();
+                return false;
+            }
 
-                return type == null || type.IsParentKind(SyntaxKind.CompilationUnit);
+            if (token.Parent.Kind() != SyntaxKind.AttributeList)
+            {
+                return true;
```

Comment:
this seems super weird.  Won't this end up showing 'assembly' after any '[' that isn't in an attribute list?  

Can we actually add a test for that to make sure that doesn't happen?

Reply:
You are right, it is weird, I made more  some test and I saw my logic was wrong, I change it.

I add that test you suggest
`class C { void Goo(string[] array) { array[$$ ...`